integration/src/main/java/com/aws/greengrass/mqtt/moquette/ClientDeviceAuthorizer.java

```diff
@@ -62,20 +63,43 @@ public boolean checkValid(String clientId, String username, byte[] password) {
                     + "client ID and has a valid AWS IoT client certificate.");
             return false;
         }
-
-        boolean canConnect = canDevicePerform(sessionId, "mqtt:connect", "mqtt:clientId:" + clientId);
+        boolean canConnect = false;
+        try {
+            canConnect = canDevicePerform(sessionId, "mqtt:connect", "mqtt:clientId:" + clientId);
+        } catch (InvalidSessionException e) {
+            LOG.error("{}: {}", e.getClass().getSimpleName(), e.getMessage());
+            try {
+                sessionPair = getOrCreateSessionForClient(clientId, username);
+                sessionId = sessionPair.getSession();
+            } catch (AuthenticationException err) {
+                LOG.atWarn().cause(err).kv(CLIENT_ID, clientId)
+                    .log("Can't create auth session. Check that the thing connects using its thing name as the "
+                        + "client ID and has a valid AWS IoT client certificate.");
+                return false;
+            }
+            try {
+                canConnect = canDevicePerform(sessionId, "mqtt:connect", "mqtt:clientId:" + clientId);
+            } catch (InvalidSessionException err) {
+                LOG.error("{}: {}", err.getClass().getSimpleName(), err.getMessage());
```

Comment:
I wouldn't bother handling this separately down here, just use the `AuthorizationException` to catch it.

Reply:
^ any reply to this?

Reply:
I was thinking that since we changed the first attempt to log at the debug level and not the error level, leaving this `InvalidSessionException` catch would allow us to log an error on the second retry. Currently, `canDevicePerform()` only throws an `InvalidSessionException` so we could catch it with the `AuthorizationException`, but if `AuthorizationException` is thrown in the future then we wouldn't have separate log statements for the different exceptions. For now though, only catching the `AuthorizationException`and logging the error would work.